### PR TITLE
fix: allow empty only fix: actually allowed an empty value before a value was selected once, and not reset

### LINKED ISSUE #10941: "Allow Empty Value" only works before you select then unselect something from the array list
### Q&A (please complete the following information)
 - OS: ALL
 - Browser: ALL
 - Version: ALL
 - Method of installation: ALL
 - Swagger-UI version:Latest/5.32.8
 - Swagger/OpenAPI version: Swagger 2 OpenAPI 3.1

### Content & configuration
Allow empty only works once. If you select something from the list, and unselect, the empty query parameter does not get attached.

Example Swagger/OpenAPI definition:
```
{
                        "name": "status",
                        "in": "query",
                        "description": "Find Something. Send status= to clear the status filter.",
                        "style": "form",
                        "explode": true,
                        "allowEmptyValue": true,
                        "schema": {
                            "anyOf": [
                                {
                                    "type": "array",
                                    "items": {
                                        "type": "string",
                                        "description": "Something",
                                        "enum": [
                                            "FOO",
                                            "BAR"
                                        ]
                                    }
                                },
                                {
                                    "type": "null"
                                }
                                
                            ]
                        }
                    },
```

Swagger-UI configuration options:
Defaults.

### To reproduce...

1. Have an api-doc similar/including to what I shared above.
2. Check the "Allow Empty" checkbox
3. Send a request and see that it works / `status=` is sent.
4. Select then unselect any item in the list. It should be empty again.
5. See that `status=` is NOT sent.

### Expected behavior
`status=` should continue to be sent after selecting/unselecting everything.

### Screenshots

Image

### Additional context or thoughts
I have already fixed this here: https://github.com/swagger-api/swagger-ui/pull/10940 :)

### PR DESCRIPTION
**Pull request by tthornton3-chwy**
### Description
The allow empty option works if you had never selected a value in the array selector widget. As soon as you selected, and unselected, something though, it would never send ?state= again. Only a refresh would allow this to work again.

fixes: #10941

### Motivation and Context

This change fixes an incoming issue where allow empty does not function properly after a value for the array is selected / unselected once.

I will create an issue and tag it momentarily.

### How Has This Been Tested?

I have tested this locally, as well as included tests in the PR. 

I used an api spec similar to this:
``` json
{
                        "name": "status",
                        "in": "query",
                        "description": "Find Something.",
                        "style": "form",
                        "explode": true,
                        "allowEmptyValue": true,
                        "schema": {
                            "anyOf": [
                                {
                                    "type": "array",
                                    "items": {
                                        "type": "string",
                                        "description": "Something.",
                                        "enum": [
                                            "FOO",
                                            "BAR"
                                        ]
                                    }
                                },
                                {
                                    "type": "null"
                                }
                                
                            ]
                        }
                    },
```
it functioned properly after my fix.

### Screenshots (if appropriate):
Screenshot 2026-06-24 at 9 28 32 AM

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/components/parameter-row.jsx [hunk 1]
```diff
@@ -147,6 +147,23 @@ describe("<ParameterRow/>", () => {
     expect(wrapper.find(".parameter__type").length).toEqual(1)
     expect(wrapper.find(".parameter__type").text()).toEqual("boolean")
   })
+
+  it("coerces empty array parameter values to null", () => {
```
Comment (glowcloud) [inline test/unit/components/parameter-row.jsx:151]: Could you replace this unit test with an e2e test (located in `test/e2e-cypress/e2e/features`) covering: 
1. Checking `Send empty value` and executing the request before selecting anything
2. Checking `Send empty value` and executing the request after selecting and deselecting a value?

**Review by glowcloud — COMMENTED** [review]
Hi @tthornton3-chwy, thanks for the PR! 

I left one request regarding tests and I also noticed a small bug with the current solution - if `Send empty value` is checked first and then `--` selected, the request will still send the empty value but the checkbox will be disabled. I would assume that it should be possible to uncheck `Send empty value` while `--` is selected.

Image
